map/src/context/RoutingManager.js

```diff
@@ -102,60 +103,439 @@ function filterMode(data) {
+    convert OSRM waypoints[] to features[] "Point" (OSRM's start-end)
+    convert OSRM duration/distance to properties
+
+    now support only 1st of routes[]
+*/
+    const features = [];
+
+    const cap = (s) => s && s[0].toUpperCase() + s.slice(1);
+
+    const maneuver = (s) => {
+        // imperative
+        const type = cap(s?.maneuver?.type ?? ''); // Turn
+        const modifier = cap(s?.maneuver?.modifier ?? ''); // Left
+
+        // target
+        const name = s?.name ?? ''; // street
+        const ref = s?.ref ? ` (${s.ref})` : ''; // (REF)
+
+        // go
+        const distance = s?.distance ?? '';
+
+        const imperative = type + ' ' + modifier;
+        const target = name ? 'to ' + name + ' ' + ref : '';
+        const go = distance > 0 ? 'and go ' + distance + ' meters' : '';
+
+        return `${imperative} ${target} ${go}`; // Turn Slight right to Schirmschneise and go 621 meters
+    }
+
+    osrm?.routes?.forEach(r => {
+        // parse geo
+        features.push({
+            type: 'Feature',
+            geometry: r.geometry,
+            properties: {
+                overall: {
+                    time: r.duration,
+                    distance: r.distance
+                }
+            }
+        });
+        // parse turns
+        r.legs?.forEach(l => {
+            l.steps?.forEach(s => {
+                features.push({
+                    type: 'Feature',
+                    geometry: {
+                        type: 'Point',
+                        coordinates: s.maneuver?.location
+                    },
+                    properties: {
+                        description: maneuver(s)
+                    }
+                });
+            })
+        });
+    });
+
+    // parse points (really need?)
+    osrm?.waypoints?.forEach(w => {
+        features.push({
+            type: 'Feature',
+            geometry: {
+                type: 'Point',
+                coordinates: w.location
+            },
+            properties: {
+                description: w.name
+            }
+        });
+    });
+
+    return { type: 'FeatureCollection', features };
+}
+
+async function calculateRoute({
+    routeProviders,
+    startPoint,
+    endPoint,
+    interPoints,
+    avoidRoads,
+    routeMode,
+    setRouteData,
+    getRouteText,
+    setRoutingErrorMsg
+}) {
+    // OsmAnd (call original function)
+    if (routeProviders.type === 'osmand') {
+        return calculateRouteOsmAnd({
+            startPoint,
+            endPoint,
+            interPoints,
+            avoidRoads,
+            routeMode,
+            setRouteData,
+            getRouteText,
+            setRoutingErrorMsg
+        });
+    }
+
+    // OSRM
+    const url = routeProviders.getURL();
+    const tail = '?geometries=geojson&overview=full&steps=true';
+
+    const points = [];
+
+    const geo = (point) => point.lng.toFixed(6) + ',' + point.lat.toFixed(6); // OSRM: lng first, lat second !
```

Comment:
it is obvious from code

Reply:
Refused: this comment emphasises that OSRM uses lng+lat, NOT usual lat+lng and this requirment isn't obvious from the code.